x-pack/legacy/plugins/lens/public/indexpattern_plugin/indexpattern_suggestions.tsx

```diff
@@ -298,17 +298,73 @@ function createNewLayerWithMetricAggregation(
 export function getDatasourceSuggestionsFromCurrentState(
   state: IndexPatternPrivateState
 ): Array<DatasourceSuggestion<IndexPatternPrivateState>> {
-  const layers = Object.entries(state.layers);
+  const layers = Object.entries(state.layers || {});
 
-  return layers
-    .map(([layerId, layer], index) => {
+  return _.flatten(
+    layers.map(([layerId, layer], index) => {
       if (layer.columnOrder.length === 0) {
-        return;
+        return [];
       }
 
-      return buildSuggestion({ state, layerId, isMultiRow: true, datasourceSuggestionId: index });
+      const onlyMetric = layer.columnOrder.every(columnId => !layer.columns[columnId].isBucketed);
+      const onlyBucket = layer.columnOrder.every(columnId => layer.columns[columnId].isBucketed);
+      if (onlyMetric || onlyBucket) {
+        // intermediary chart, don't try to suggest reduced versions
+        return [
+          buildSuggestion({ state, layerId, isMultiRow: false, datasourceSuggestionId: index }),
+        ];
+      }
+
+      return createSimplifiedTableSuggestions(state, layerId);
+    })
+  ).map(
+    (suggestion, index): DatasourceSuggestion<IndexPatternPrivateState> => ({
+      ...suggestion,
+      table: { ...suggestion.table, datasourceSuggestionId: index },
```

Comment:
This is the second time I've seen this .map(...) logic used to assign datasourceSuggestionId in this PR. I wonder if it's worth pulling out into a helper as in my code snippet above? Probably not? Just calling it to your attention.